src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -291,6 +445,56 @@ private UInt256 GetNonce(Keccak stateRoot, Address address)
             return _processingEnv.StateReader.GetNonce(stateRoot, address);
         }
 
+        //Apply changes to accounts and contracts states including precompiles
+        private void ModifyAccounts(AccountOverride[] StateOverrides, IWorldState? StateProvider, IReleaseSpec? CurrentSpec)
+        {
+            Account? acc;
+
+            foreach (AccountOverride accountOverride in StateOverrides)
+            {
+                Address address = accountOverride.Address;
+                bool accExists = StateProvider.AccountExists(address);
+                if (!accExists)
+                {
+                    StateProvider.CreateAccount(address, accountOverride.Balance, accountOverride.Nonce);
+                    acc = StateProvider.GetAccount(address);
+                }
+                else
+                    acc = StateProvider.GetAccount(address);
+
+                UInt256 accBalance = acc.Balance;
+                if (accBalance > accountOverride.Balance)
+                    StateProvider.SubtractFromBalance(address, accBalance - accountOverride.Balance, CurrentSpec);
+                else if (accBalance < accountOverride.Nonce)
+                    StateProvider.AddToBalance(address, accountOverride.Balance - accBalance, CurrentSpec);
+
+                UInt256 accNonce = acc.Nonce;
+                if (accNonce > accountOverride.Nonce)
+                    StateProvider.DecrementNonce(address);
+                else if (accNonce < accountOverride.Nonce) StateProvider.IncrementNonce(address);
```

Comment:
This should be refactored to one method that tries to GetAccount, sets it values and sets accont. If it is empty creates new account with correct values.

Reply:
More readable functions extracted for GettingAccount, Balance, Nonce, Code, State